sw/host/opentitanlib/src/test_utils/lc_transition.rs

```diff
@@ -129,28 +128,38 @@ fn setup_lc_transition(
 /// transport.reset_target(init.bootstrap.options.reset_delay, true).unwrap();
 ///
 /// // Connect to the LC controller TAP.
-/// let jtag = transport.jtag(jtag_opts).unwrap();
-/// jtag.connect(JtagTap::LcTap).expect("failed to connect to LC TAP");
+/// let mut jtag = transport
+///     .jtag(jtag_opts)
+///     .unwrap()
+///     .connect(JtagTap::LcTap)
+///     .expect("failed to connect to LC TAP");
 ///
 /// let test_exit_token = DifLcCtrlToken::from([0xff; 16]);
 ///
 /// lc_transition::trigger_lc_transition(
 ///     &transport,
-///     jtag.clone(),
+///     jtag,
 ///     DifLcCtrlState::Prod,
 ///     Some(test_exit_token.into_register_values()),
 ///     true,
 ///     init.bootstrap.options.reset_delay,
+///     Some(JtagTap::LcTap),
 /// ).expect("failed to trigger transition to prod");
 ///
+/// jtag = transport
+///     .jtag(jtag_opts)
+///     .unwrap()
+///     .connect(JtagTap::LcTap)
+///     .expect("failed to reconnect to LC TAP");
+///
 /// assert_eq!(
 ///     jtag.read_lc_ctrl_reg(&LcCtrlReg::LCState).unwrap(),
 ///     DifLcCtrlState::Prod.redundant_encoding(),
 /// );
 /// ```
 pub fn trigger_lc_transition(
```

Comment:
Since you changed `trigger_lc_transition` to consume the jtag tap, it's quite confusing to have a `reconnect_jtag_tap` parameters. I think it should at least be renamed, maybe `new_tap_straps` or `reset_tap_straps`?
I understand your rationale for consuming the tap, the downside is that everytime we use this function, we have to recreate the tap manually. Would it be so bad if this function optionally recreate it for you and returns it?

Reply:
It wouldn't be bad if the function always recreate it; see `trigger_volatile_raw_unlock`. But since this function optionally recreate it, it needs to return `Result<Option<Box<dyn Jtag + 't>>>` and user has to `unwrap` it, which IMO is more ugly than just let caller reconnect.